ext/srt/gstsrtobject.c

```diff
@@ -253,6 +254,22 @@ gst_srt_object_set_common_params (SRTSOCKET sock, GstSRTObject * srtobject,
     }
   }
 
+  if (gst_structure_has_field (srtobject->parameters, "buffer-size")) {
+    gint32 buffer_size;
+
+    gst_structure_get_int (srtobject->parameters, "buffer-size", &buffer_size);
+
+    if (buffer_size > 0) {
+      gboolean is_sender = GST_IS_BASE_SINK (srtobject->element);
+      SRT_SOCKOPT opt = is_sender ? SRTO_SNDBUF : SRTO_RCVBUF;
+
+      if (srt_setsockopt (sock, 0, opt, &buffer_size, sizeof (gint32))) {
+        g_set_error (error, GST_LIBRARY_ERROR, GST_LIBRARY_ERROR_SETTINGS,
+            "failed to set buffer size (reason: %s)", srt_getlasterror_str ());
+      }
+    }
+  }
+
```

Comment:
This change should be enough for our purpose. However, it's very dubious whether we can keep going with this approach for the rest of the SRT socket options.

I'd suggest the setting approach, `option-string`,  of x265enc. That way probably is more convenient for users and programmers.

Reply:
That's possible, but implementing such generic means to set any socket property will take additional 1-2 days of effort and I think it's better handled as a separate task. Please open a new ticket if you want us to work on it.